reviews/graduation-envoy.md

```diff
@@ -0,0 +1,51 @@
+# Envoy Graduation Application
+
+Envoy was OSSd in September, 2016 and joined the CNCF in September, 2017. The project has grown
+significantly over time.
+
+The following application links to the required information to become a graduated project.
+
+## Envoy fulfills all the incubating and graduation criteria:
+
+### Document that it is being used successfully in production by at least three independent end users which, in the TOC’s judgement, are of adequate quality and scope.
+
+* Known end users include: AirBnb, Alibaba, Apple, Booking.com, Cookpad, DigitalOcean, eBay, Google, GO-JEK,
```

Comment:
Please clarify whether all of these companies use Envoy in *production*.  Also, please provide some more detail on the production usage for at least three of the most major uses, e.g. exactly what product/service uses Envoy, indication of scale of usage (both in terms of infrastructure size and number of users), how long it has been used in production etc.  This will make it easier to judge the "quality and scope" aspect of the requirement.

Reply:
Many/most major users of Envoy are not forthcoming about exactly how they use it and at what scale. Although I know that almost all of the names listed here are in production, some at very large scale, I don't have permission to speak on their behalf. The only one I can speak officially on is Lyft.

@caniszczyk thoughts on best way to proceed here?

Reply:
@mattklein123 as a start, I would ask some adopters to see if they are comfortable publicly stating their usage of Envoy on this PR in more detail

Reply:
Speaking for the Lyft side of things:
- We run > 20K Envoy instances at once.
- 100% of edge and StS traffic flows through Envoy.
- The great majority of external partner traffic flows through Envoy.
- All MongoDB, Dynamo, and Redis traffic flows through Envoy.
- Total mesh traffic is 10M+ HTTP/gRPC RPS. Redis query RPS is many multiples of that.

@caniszczyk is going to help with the legwork to see what different companies are willing to say publicly.

Reply:
I'm not sure [Cookpad](https://www.cookpadteam.com/) case is major one though, we can share our usage of Envoy:

- [Our recipe sharing service](https://cookpad.com/) and some of other services are using Envoy. The Japanese recipe sharing service has more than 55 millions of average monthly users.
- 700~1200 Envoy instances are running in production.
- We use Envoy in production since November 2017.
- Most (more than 3/4) of our internal (service-to-service) traffics flow through Envoy.

Reply:
AppDirect is using Envoy (via Ambassador) and routing 100% of production traffic: https://www.appdirect.com/blog/evolution-of-the-appdirect-kubernetes-network-infrastructure.

Reply:
Pinterest is using Envoy for 100% of edge traffic and we're actively working on service mesh.

Reply:
At Pinterest, we have deployed Envoy as our edge proxy, where it currently serves 250 million monthly unique users. In addition, we currently are developing a Thrift service mesh based on Envoy.

Reply:
Coursera is using Envoy in production. All requests to backend services flow through Envoy. We have more than 450 Envoy instances running in production, which serve about 70 backend services. Our total request flow volume is about 10k RPS.

Reply:
Salesforce uses Envoy in production for service proxies. 100% of service-service traffic flows through Envoy.

Reply:
At [GO-JEK](https://www.go-jek.com/) we are using envoy in production for more than two years now.
 We are using envoy as GRPC/HTTP loadbalancer and gate  for GOPAY wallet, all other GOJEK products talk to GOPAY WALLET(100+ microservices internally within GOPAY), external integrations as well , plus mobile apps. We serve millions of customers and partners(drivers, merchants etc) through this.
We have we been using envoy for application sharding our transaction history service, which serves millions of requests daily for both customers, drivers, merchants etc.

Reply:
Thank you all for your responses.  Collectively these adequately address my question.

Reply:
thanks all, @quinton-hoole-2 would you be comfortable kicking off a graduation vote then?

Reply:
I'm in favor of kicking off a vote.

Reply:
@caniszczyk Yes, I'm in favor of kicking off the graduation vote.  Apologies for my delay replying.

Reply:
Thanks @quinton-hoole-2, I kicked off the graduation vote: https://lists.cncf.io/g/cncf-toc/message/2633

Reply:
@quinton-hoole-2 I remembered Huawei also has effort on Envoy, right?